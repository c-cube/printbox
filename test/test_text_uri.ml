let b =
  let open PrintBox in
  vlist
    [
      link ~uri:"https://example.com/1" @@ frame @@ text "child 1";
      link ~uri:"https://example.com/2" @@ text "child 2";
      frame
      @@ tree (link ~uri:"https://example.com/3" empty)
           [ link ~uri:"https://example.com/4" @@
             tree (frame @@ text "header 3") [ frame @@ text "subchild 3" ] ];
      link ~uri:"https://example.com/5" @@
      tree empty
        [ tree (frame @@ text "header 4") [ frame @@ text "subchild 4" ] ];
      frame @@ text "child 5";
      link ~uri:"https://example.com/6" @@
      frame
      @@ tree
           (frame @@ text "header 6")
           [ tree (frame @@ text "child 6") 
             [ link ~uri:"https://example.com/7" @@ frame @@ text "subchild 6" ] ];
    ]

let () = print_endline @@ PrintBox_text.to_string b
