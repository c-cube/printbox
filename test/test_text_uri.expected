┌───────┐
│]8;;https://example.com/1\child 1]8;;\│
└───────┘
────────────────────
]8;;https://example.com/2\child 2]8;;\
────────────────────
┌──────────────────┐
│──┬────────┐      │
│  │]8;;https://example.com/4\header 3]8;;\│      │
│  ├────────┘      │
│  └─┬──────────┐  │
│    │]8;;https://example.com/4\subchild 3]8;;\│  │
│    └──────────┘  │
└──────────────────┘
────────────────────
──┬────────┐
  │]8;;https://example.com/5\header 4]8;;\│
  ├────────┘
  └─┬──────────┐
    │]8;;https://example.com/5\subchild 4]8;;\│
    └──────────┘
────────────────────
┌───────┐
│child 5│
└───────┘
────────────────────
┌──────────────────┐
│┌────────┐        │
││]8;;https://example.com/6\header 6]8;;\│        │
│├────────┘        │
│└─┬───────┐       │
│  │]8;;https://example.com/6\child 6]8;;\│       │
│  ├───────┘       │
│  └─┬──────────┐  │
│    │]8;;https://example.com/7\subchild 6]8;;\│  │
│    └──────────┘  │
└──────────────────┘
